dvc/utils/stream.py

```diff
@@ -61,17 +63,30 @@ def peek(self, n):
         return self.leftover[:n]
 
 
-class HashedStreamReader:
+class HashedStreamReader(io.IOBase):
```

Comment:
I wasn't expecting this to become a file-like interface (rather just a class with `.read()`) so I named it `StreamReader`. Now it seems to be not exactly fit but I don't have a strong opinion on it.